### PR TITLE
Add documentation for loadPlugin()

### PR DESCRIPTION
**Pull request by markstory**
Refs cakephp/cakephp#10711

### CODE REVIEW

File: en/plugins.rst [hunk 1]
```diff
@@ -280,6 +280,54 @@ autoloader once you've created your plugin:
...
+
+    <?php
+    use Cake\Routing\Route\DashedRoute;
+    use Cake\Routing\Router;
+
+    Router::plugin(
+        'ContactManager',
+        ['path' => '/contact-manager'],
+        function ($routes) {
+            $routes->fallbacks(DashedRoute::class);
```
Comment (ravage84) [inline en/plugins.rst]: May be add one plugin specific rule plus the fallbacks? Might help envision what one can do with it.

File: en/plugins.rst [hunk 2]
```diff
@@ -280,6 +280,54 @@ autoloader once you've created your plugin:
...
+        }
+    );
+
+The above will connect default routes for you plugin. You can customize this
```
Comment (ravage84) [inline en/plugins.rst]: If we add another route, this would need to change a bit.

File: en/plugins.rst [hunk 3]
```diff
@@ -280,6 +280,54 @@ autoloader once you've created your plugin:
...
+
+    Plugin::load('ContactManager', ['routes' => true]);
+
+You can also load plugin routes in your application's routes list. Doing this
+provides you more control on how plugin routes are loaded and allows you to wrap
+plugin routes in additional scopes or prefixes::
+
+    Router::scope('/', function ($routes) {
+        // Connect other routes.
+        $routes->scope('/a', function ($routes) {
```
Comment (ravage84) [inline en/plugins.rst]: may be not ``/a``...

File: en/plugins.rst [hunk 4]
```diff
@@ -280,6 +280,54 @@ autoloader once you've created your plugin:
...
+            $routes->loadPlugin('ContactManager');
+        });
+    });
+
+The above would result in URLs like ``/a/contact_manager/contacts``.
```
Comment (ravage84) [inline en/plugins.rst]: ``/a`` ...

File: en/plugins.rst [hunk 5]
```diff
@@ -319,34 +367,6 @@ Also make the ``AppController`` if you don't have one already::
...
 
-Before you can access your controllers, you'll need to ensure the plugin is
-loaded and the plugin routes are loaded.  In your **config/bootstrap.php** add
-the following::
-
-    Plugin::load('ContactManager', ['routes' => true]);
-
-If you are using ``Plugin::loadAll()`` ensure that routes are loaded::
-
-    Plugin::loadAll(['routes' => true]);
```
Comment (ravage84) [inline en/plugins.rst]: So, we do not recommend this anymore?

File: en/plugins.rst (hunk 2 above)
Comment (ravage84) [inline en/plugins.rst]: "for youR plugin"

File: en/appendices/3-5-migration-guide.rst [hunk 6]
```diff
@@ -180,6 +180,7 @@ Routing
   match.
 * ``Cake\Routing\RouteBuilder`` now has methods to create routes for
   specific HTTP methods. e.g ``get()`` and ``post()``.
+* ``Cake\Routing\RouteBuilder::loadPlugin()`` was added.
```
Comment (markstory) [inline en/appendices/3-5-migration-guide.rst]: I'm thinking we don't deprecate it based on the discussion we had about making plugin routes more complicated.

File: en/plugins.rst (hunk 1 above)
Comment (markstory) [inline en/plugins.rst]: Done.

File: en/plugins.rst (hunk 3 above)
Comment (markstory) [inline en/plugins.rst]: What would you suggest?

File: en/plugins.rst (hunk 5 above)
Comment (markstory) [inline en/plugins.rst]: I hate loadAll() personally.

File: en/appendices/3-5-migration-guide.rst (hunk 6 above)
Comment (ravage84) [inline en/appendices/3-5-migration-guide.rst]: Fine, but then https://github.com/cakephp/cakephp/pull/10711/files#diff-7997caf44fa801d62e790043f405bd34R332 needs to be removed.

File: en/plugins.rst (hunk 3 above)
Comment (ravage84) [inline en/plugins.rst]: May be a locale, like ``de_DE`` or may be ``api`` as it was used on https://book.cakephp.org/3.0/en/development/routing.html#using-named-routes.

File: en/appendices/3-5-migration-guide.rst (hunk 6 above)
Comment (ravage84) [inline en/appendices/3-5-migration-guide.rst]: Fine, but https://github.com/cakephp/cakephp/pull/10711/files#diff-7997caf44fa801d62e790043f405bd34R332 needs to be removed then.

File: en/plugins.rst (hunk 5 above)
Comment (ravage84) [inline en/plugins.rst]: Fine with me, as it is mentioned in other places on that page, anyway.

File: en/appendices/3-5-migration-guide.rst (hunk 6 above)
Comment (markstory) [inline en/appendices/3-5-migration-guide.rst]: Good point. Thank you.

File: en/plugins.rst (hunk 3 above)
Comment (markstory) [inline en/plugins.rst]: Added a `/backend` path segment.

**Pull request merged**